[No QA] Update to Webpack 5 and peer dependencies

afmire877 opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing.
Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the issue; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ [#10033](https://github.com/Expensify/App/issues/10033)

### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Verify you can install app dependencies and that no errors appear. (npm install)
1. Verify you can still run the web locally and use the app normally with no issues. (npm run web)

- [x] Verify that no errors appear in the JS console

### PR Review Checklist
<!--
This is a checklist for PR authors & reviewers. Please make sure to complete all tasks and check them off once you do, or else Expensify has the right not to merge your PR!
-->
#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps
Marked as No QA because we will know if this worked or not based on:

1. Deploy workflows
1. Regression tests.

There's nothing specific to QA here.

- [ ] Verify that no errors appear in the JS console

### Screenshots
<!-- Add screenshots for all platforms tested. Pull requests won't be merged unless the screenshots show the app was tested on all platforms.-->
![2022-08-02 17 08 37](https://user-images.githubusercontent.com/33981371/182422714-123784b3-970c-4816-99d9-6bab95581e55.gif)



#### Web
<!-- Insert screenshots of your changes on the web platform-->

#### Mobile Web
<!-- Insert screenshots of your changes on the web platform (from a mobile browser)-->

#### Desktop
<!-- Insert screenshots of your changes on the desktop platform-->
![2022-08-02 17 59 12](https://user-images.githubusercontent.com/33981371/183229280-8e65d971-8158-422c-955e-01b559c87dfe.gif)

#### iOS
<!-- Insert screenshots of your changes on the iOS platform-->
<img width="1273" alt="Screenshot 2022-08-06 at 03 06 59" src="https://user-images.githubusercontent.com/33981371/183229420-2fd09b93-d92c-437b-93f3-0e7318130f1b.png">


#### Android
<!-- Insert screenshots of your changes on the Android platform-->

afmire877 committed:
upgraded all pkgs to webpack 5

upgraded all pkgs to webpack 5

fix: changed import for all named export for package.json

chore: update to latest version webpack-dev-server

revert devServer.hot

update env flgs

fix

updated @react-navigation/drawer

fix: package.json

revert: @react-navigation/drawer

npm install

roryabraham commented on src/setup/platformSetup/index.website.js:
```diff
@@ -6,7 +6,7 @@ import 'shim-keyboard-event-key';
 import checkForUpdates from '../../libs/checkForUpdates';
 import Config from '../../CONFIG';
 import DateUtils from '../../libs/DateUtils';
-import {version as currentVersion} from '../../../package.json';
```
Is this change necessary?

afmire877 replied:
Webpack will only allow ESM to have named imports. So non-ESM can only be doing 'default' imports 

[Here is the reference from Webpack 5 docs](https://webpack.js.org/guides/ecma-script-modules/#flagging-modules-as-esm:~:text=Only%20the%20%22default%22%20export%20can%20be%20imported%20from%20non%2DESM.%20Named%20exports%20are%20not%20available)

<img width="1309" alt="Screenshot 2022-08-08 at 22 11 44" src="https://user-images.githubusercontent.com/33981371/183518048-6d0bdbda-829f-49a3-89e8-58303f20535e.png">

roryabraham commented on src/pages/settings/AboutPage/AboutPage.js:
```diff
@@ -13,7 +13,7 @@ import withLocalize, {withLocalizePropTypes} from '../../../components/withLocal
 import MenuItem from '../../../components/MenuItem';
 import Logo from '../../../../assets/images/new-expensify.svg';
-import {version} from '../../../../package.json';
+import pkg from '../../../../package.json';
```
Is this change necessary?

afmire877 replied:
Same as https://github.com/Expensify/App/pull/10286?notification_referrer_id=NT_kwDOAgaDu7M0MTQ2ODUxMDIyOjMzOTgxMzcx#discussion_r940681238

roryabraham commented on desktop/start.js:
```diff
@@ -8,7 +8,7 @@ const basePort = 8080;
     port: basePort,
 }).then((port) => {
-    const devServer = `webpack-dev-server --config config/webpack/webpack.dev.js --port ${port} --env.platform desktop`;
+    const devServer = `webpack-dev-server --config config/webpack/webpack.dev.js --port ${port} --env platform=desktop`;
```
Is this change necessary?

afmire877 replied:
In Webpack v5, syntax on env variables has changed to be `--env  key=value`


Reference: https://webpack.js.org/guides/environment-variables/#root

roryabraham reviewed (changes requested):
Overall it looks much better, but I suspect this PR contains a number of unnecessary changes:

1. Getting rid of destructured imports of `package.json` doesn't seem necessary.
1. Replacing `--env.platform=desktop` with `--env platform=desktop` doesn't seem necessary either.

Please review this PR and remove any unnecessary and/or unrelated changes. Thanks!

afmire877 commented on src/libs/API.js:
```diff
@@ -73,7 +73,7 @@ function makeRequestWithSideEffects(command, apiCommandParameters = {}, onyxData
     const data = {
         ...apiCommandParameters,
-        appversion: version,
+        appversion: pkg.version,
```
Same as https://github.com/Expensify/App/pull/10286?notification_referrer_id=NT_kwDOAgaDu7M0MTQ2ODUxMDIyOjMzOTgxMzcx#discussion_r940681238

afmire877 commented on src/libs/Environment/betaChecker/index.android.js:
```diff
@@ -1,5 +1,5 @@
 import CONST from '../../../CONST';
-import {version} from '../../../../package.json';
+import pkg from '../../../../package.json';
```
Same as https://github.com/Expensify/App/pull/10286?notification_referrer_id=NT_kwDOAgaDu7M0MTQ2ODUxMDIyOjMzOTgxMzcx#discussion_r940681238

afmire877 commented on config/webpack/webpack.common.js:
```diff
@@ -45,6 +45,9 @@ const webpackConfig = ({envFile = '.env', platform = 'web'}) => ({
             usePolyfillIO: platform === 'web',
         }),
+        new ProvidePlugin({
+            process: 'process/browser',
```
`process` has removed the polyfill for this Node.js variable. This variable is still required at runtime by packages like `react-native-reanimated`. 

Here is the error we get.

<img width="1313" alt="Screenshot 2022-08-08 at 23 06 20" src="https://user-images.githubusercontent.com/33981371/183523565-247a4fe8-3ec6-43d0-92b2-a3374f72e5ba.png">


Doc reference: https://webpack.js.org/migrate/5/#:~:text=process%20is%20not%20defined

roryabraham commented:
Thanks for the links @afmire877, it's very helpful to understand the context for why those changes were needed. Turns out I was prettymuch wrong on all counts with my previous review 🙃

roryabraham commented:
You'll need to [retroactively GPG-sign these commits](https://webdevstudios.com/2020/05/26/retroactively-sign-git-commits/) before we'll be able to approve and merge this, however

afmire877 commented:
I have added one more commit to update `@react-navigation/drawer` to fix the warning below.

<img width="971" alt="Screenshot 2022-08-09 at 00 46 13" src="https://user-images.githubusercontent.com/33981371/183767298-514472ae-3339-4c10-a3f4-1d6bdb453aa8.png">

afmire877 commented on package.json:
```diff
@@ -74,6 +74,7 @@
     "moment": "^2.29.4",
     "moment-timezone": "^0.5.31",
     "onfido-sdk-ui": "^8.1.0",
+    "process": "^0.11.10",
```
Same as https://github.com/Expensify/App/pull/10286#discussion_r940708335

afmire877 commented on package.json:
```diff
@@ -55,7 +55,7 @@
     "@react-native-firebase/perf": "^12.3.0",
     "@react-native-picker/picker": "^2.3.1",
-    "@react-navigation/drawer": "6.3.0",
+    "@react-navigation/drawer": "^6.4.3",
```
Nope. The build is not failing so I will revert 👍🏾

afmire877 commented:
Ready for re-review 👀  @parasharrajat @roryabraham

afmire877 committed:
upgraded all pkgs to webpack 5

upgraded all pkgs to webpack 5

fix: changed import for all named export for package.json

chore: update to latest version webpack-dev-server

revert devServer.hot

update env flgs

fix

updated @react-navigation/drawer

fix: package.json

revert: @react-navigation/drawer

npm install

npm install

npm install

fix package.json

fix: package.json

package.json fix

afmire877 commented:
@parasharrajat  👍🏾  done a rebase and npm install

roryabraham commented:
@afmire877 Sorry, you've got conflicts again.

afmire877 commented:
> @afmire877 Sorry, you've got conflicts again.

Sorted +1

roryabraham commented:
Hi @afmire877, thanks for updating conflicts again. There's a few more things to address:

1. I'm pretty suspicious of some of the changes in `package-lock.json` – it seems to have gotten out-of-sync somehow. For example, I see that the `octokit` version is changed in this PR even though you didn't change that dependency in `package.json`. That's what's causing the GitHub Actions CI failure you can see above ^. Did you delete `package-lock.json` at some point? I think the easiest way to fix this might be to:
    1. [copy the contents of `package-lock.json` on the upstream `main`](https://raw.githubusercontent.com/Expensify/App/main/package-lock.json)
    1. paste that into your local `package-lock.json`
    1. run `npm install` again
    1. Commit the diff in your `package-lock.json` to this PR
1. Also, please run `npx browserslist@latest --update-db` and commit the diff in the lockfile

roryabraham commented:
@afmire877 Did you see my last comment? I don't want us to commit the changes to octokit because those are out-of-scope and it's a sign that this lockfile is incorrect.

afmire877 commented:
Yeah just saw after i commeted. I will address it now

afmire877 commented:
Done @roryabraham ✅

roryabraham commented:
Very sorry, there are conflicts again. I'll try to review this one faster.

afmire877 commented:
@roryabraham no problem. Rebased again

roryabraham commented:
Looks like we've got conflicts, lint errors, and unit test failures

roryabraham commented:
@afmire877 I'm not sure how it's related to this pull request, but I'm now seeing some new console errors:

<img width="1716" alt="image" src="https://user-images.githubusercontent.com/47436092/186093449-9811c86b-0dce-4a46-9c03-df6ee8b4ab10.png">

roryabraham commented:
Maybe we can try this?

```
resolve: { fallback: { 'process/browser': require.resolve('process/browser'), } }
```

roryabraham commented:
We could maybe also do [this solution](https://stackoverflow.com/questions/69427025/programmatic-webpack-jest-esm-cant-resolve-module-without-js-file-exten), but I think the first one looks preferable.

roryabraham commented:
Seems like progress has stalled here – @afmire877 when can we expect an update?

afmire877 commented:
Hey, @roryabraham I will get to you in the next few days. I apologise for the late replies.

afmire877 committed:
fix: process

roryabraham reviewed (approved):
- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Additional review notes
- There is a console warning that pops up, but it is unrelated and can be handled in a separate PR. You will notice that it takes over the screen (which it did not before), but I'm hopeful that will incentivize us to fix console warning more quickly in the future.
- There's some caching stuff I'd like to follow-up on to make webpack builds faster.
- Also testing a production build with `npm run build` and it seemed to work.

roryabraham commented:
Tests were passing

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/2991935016) to staging by @roryabraham in version: 1.1.97-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌ 
🕸 web 🕸|success ✅
